### PR TITLE
Resumes must be PDF

### LINKED ISSUE #292: Restrict Resumes to PDF
## Is this a BUG REPORT or FEATURE REQUEST?
Feature Request

## What happened
(Include screenshots when possible)
Hackers send resumes as power point, libre office documents, etc. We even had hackers that sended memes as resumes because the field is not restricted.

## What you expected to happen
From now on only accept .pdf files as Resumes. It will be easier to review as well.

## How to reproduce it (as minimally and precisely as possible)
Add a validator to the resume field from application model. And put the files extensions on hackathon variables.

## Environment
- Browser version (Ex: Google Chrome 62.0.3202.94 (64-bit)...):
- Domain: 
- Others:

(Optional)
- Python version (`python --version`)
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):

### PR DESCRIPTION
**Pull request by Casassarnau**
## What this PR does / Why we need it
Add a validator to the resume field from application model. And put the files extensions on hackathon variables.

## Which issue(s) this PR fixes (optional) 

Fixes #292

## Some questions
- [x] I have read the contributing guidelines
- [x] I abide by this repository Code of Conduct
- [x] I understand that my PR won't be merged until GitHub gives a "green light"

## Additional Notes (optional)
Do you want to add anything else? We :heart: to hear your opinions!

### CODE REVIEW

**Comment by Casassarnau** [comment]
I saw there was 2 Django vulnerabilities, so I fixed them by upgrading the django version

File: applications/models.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 from app import utils
 from user.models import User
+from .validators import validate_file_extension
```
Comment (casassg) [inline applications/models.py]: No relative path pls
```suggestion
from applications.validators import validate_file_extension
```

**Review by casassg — CHANGES_REQUESTED** [review]
Fes servir regex millor.

File: applications/forms.py [hunk 2]
```diff
@@ -216,7 +216,7 @@ class Meta:
             'origin': 'Where are you joining us from?',
             'description': 'Why are you excited about %s?' % settings.HACKATHON_NAME,
             'projects': 'What projects have you worked on?',
-            'resume': 'Upload your resume',
+            'resume': 'Upload your resume (%s)' % ', '.join(settings.SUPPORTED_RESUME_EXTENSIONS),
```
Comment (casassg) [inline applications/forms.py]: What happens if `SUPPORTED_RESUME_EXTENSIONS` is not set?

**Review by casassg — CHANGES_REQUESTED** [review]

**Comment by casassg** [comment]
I also, activated review deployments to do integration testing 😄

File: applications/forms.py (hunk 2 above)
Comment (Casassarnau) [reply applications/forms.py]: But it is, I created it. if it's empty it will show only ()

File: applications/forms.py [hunk 3]
```diff
@@ -216,7 +216,8 @@ class Meta:
             'origin': 'Where are you joining us from?',
             'description': 'Why are you excited about %s?' % settings.HACKATHON_NAME,
             'projects': 'What projects have you worked on?',
-            'resume': 'Upload your resume',
+            'resume': 'Upload your resume %s' % ('' if not settings.SUPPORTED_RESUME_EXTENSIONS
+                                                 else '(%s)' % ', '.join(settings.SUPPORTED_RESUME_EXTENSIONS)),
```
Comment (casassg) [inline applications/forms.py]: Move formats to `help_texts` better.

File: applications/forms.py [hunk 4]
```diff
@@ -216,7 +216,8 @@ class Meta:
             'origin': 'Where are you joining us from?',
             'description': 'Why are you excited about %s?' % settings.HACKATHON_NAME,
             'projects': 'What projects have you worked on?',
-            'resume': 'Upload your resume',
+            'resume': 'Upload your resume %s' % ('' if not settings.SUPPORTED_RESUME_EXTENSIONS
```
Comment (casassg) [inline applications/forms.py]: `settings.SUPPORTED_RESUME_EXTENSIONS` will fail if if `SUPPORTED_RESUME_EXTENSIONS` is not set in the settings. Use `getattr(settings, 'SUPPORTED_RESUME_EXTENSIONS', None)` instead

File: applications/forms.py [hunk 5]
```diff
@@ -196,7 +198,8 @@ class Meta:
                          ' http://textsmili.es/</a>',
             'projects': 'You can talk about about past hackathons, personal projects, awards etc. '
                         '(we love links) Show us your passion! :D',
-            'reimb_amount': 'We try our best to cover costs for all hackers, but our budget is limited'
+            'reimb_amount': 'We try our best to cover costs for all hackers, but our budget is limited',
+            'resume': 'Accepted file formats: (%s)' % ', '.join(extensions) if extensions else ''
```
Comment (casassg) [inline applications/forms.py]: `Accepted file formats: ()`?

**Review by casassg — CHANGES_REQUESTED** [review]

File: applications/forms.py [hunk 6]
```diff
@@ -199,7 +199,7 @@ class Meta:
             'projects': 'You can talk about about past hackathons, personal projects, awards etc. '
                         '(we love links) Show us your passion! :D',
             'reimb_amount': 'We try our best to cover costs for all hackers, but our budget is limited',
-            'resume': 'Accepted file formats: (%s)' % ', '.join(extensions) if extensions else ''
+            'resume': 'Accepted file formats: %s' % ', '.join(extensions) if extensions else ''
```
Comment (casassg) [inline applications/forms.py]:
```suggestion
            'resume': 'Accepted file formats: %s' % ', '.join(extensions) if extensions else 'Any'
```

**Review by casassg — CHANGES_REQUESTED** [review]

File: applications/forms.py (hunk 6 above)
Comment (Casassarnau) [reply applications/forms.py]: No funcionara si poses aixo

Comment (Casassarnau) [reply applications/forms.py]: Has de posar parentesis

Comment (Casassarnau) [reply applications/forms.py]: Treu la sugestion i faig un altre commit

**Review by casassg — APPROVED** [review]

**Pull request merged**